src/Sulu/Bundle/ContactBundle/Resources/js/containers/ContactAccountSelection/ContactAccountSelection.js

```diff
@@ -0,0 +1,158 @@
+import {action, computed, observable, toJS} from 'mobx';
+import equals from 'fast-deep-equal';
+import {MultiItemSelection} from 'sulu-admin-bundle/components';
+import {MultiListOverlay} from 'sulu-admin-bundle/containers';
+import {translate} from 'sulu-admin-bundle/utils';
+import ContactAccountSelectionStore from './stores/ContactAccountSelectionStore';
+import contactAccountSelectionStyles from './contactAccountSelection.scss';
+
+type Props = {|
+    disabled: boolean,
+    onChange: (value: Array<Object>) => void,
+    value: Array<Object>,
+|};
+
+@observer
+class ContactAccountSelection extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+        value: [],
+    };
+
+    @observable openedOverlay: ?string = undefined;
+    store: ContactAccountSelectionStore;
+
+    constructor(props: Props) {
+        super(props);
+
+        const {value} = this.props;
+
+        this.store = new ContactAccountSelectionStore();
+        this.store.loadItems(value);
+    }
+
+    componentDidUpdate() {
+        const {value} = this.props;
+
+        const newIds = toJS(value);
+        const loadedIds = this.loadedIds;
+
+        newIds.sort();
+        loadedIds.sort();
+        if (!equals(newIds, loadedIds) && !this.store.loading) {
+            this.store.loadItems(value);
+        }
+    }
+
+    @computed get loadedIds(): Array<string> {
+        return toJS(this.store.items.map((item) => item.id));
+    }
+
+    @action handleAddButtonClick = (type: ?string) => {
+        this.openedOverlay = type;
+    };
+
+    @action handleOverlayClose = () => {
+        this.openedOverlay = undefined;
+    };
+
+    @action handleConfirm(items: Array<Object>, prefix: string) {
+        const {onChange, value} = this.props;
+
+        const itemIds = items.map((contact) => prefix + contact.id);
+
+        onChange([
+            ...value.filter((id) => !id.startsWith(prefix) || itemIds.includes(id)),
+            ...itemIds,
+        ].filter((item, index, items) => index == items.indexOf(item)));
+
+        this.openedOverlay = undefined;
+    }
+
+    @action handleContactConfirm = (contacts: Array<Object>) => {
+        this.handleConfirm(contacts, ContactAccountSelectionStore.contactPrefix);
+    };
+
+    @action handleAccountConfirm = (accounts: Array<Object>) => {
+        this.handleConfirm(accounts, ContactAccountSelectionStore.accountPrefix);
+    };
+
+    callChange() {
+        const {onChange} = this.props;
+
+        onChange(this.loadedIds);
+    }
+
+    handleRemove = (id: string) => {
+        this.store.remove(id);
+        this.callChange();
+    };
+
+    handleSorted = (oldItemIndex: number, newItemIndex: number) => {
+        this.store.move(oldItemIndex, newItemIndex);
+        this.callChange();
+    };
+
+    render() {
+        const {disabled, value} = this.props;
+
+        return (
+            <Fragment>
+                <MultiItemSelection
+                    disabled={disabled || false}
```

Comment:
should not be needed but 🤷‍♂

Reply:
IIRC it is needed, because the `MultiItemSelection` component only accepts a boolean, and `disabled` could also be `undefined`.

Reply:
you defined the type of the `disabled` prop of this component as non nullable `boolean` too, but i dont think this will lead to any problems 😉